src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -1447,6 +1452,97 @@ public abstract sealed class $Type$Buffer
         return put(src, 0, src.length());
     }
 
+    /**
+     * Absolute bulk <i>put</i> method&nbsp;&nbsp;<i>(optional operation)</i>.
+     *
+     * <p> This method transfers $type$s from the given
+     * {@linkplain CharSequence} into this buffer.  If there are more $type$s
+     * to be copied from the {@code CharSequence} than remain in this buffer,
+     * that is, if
+     * <code>end&nbsp;-&nbsp;start</code>&nbsp;{@code >}&nbsp;
+     * {@code limit()}&nbsp;-&nbsp;{@code index},
+     * then no $type$s are transferred and a {@link BufferOverflowException}
+     * is thrown.
+     *
+     * <p> Otherwise, this method copies
+     * <i>n</i>&nbsp;=&nbsp;{@code end}&nbsp;-&nbsp;{@code start} $type$s
+     * from the given {@code CharSequence} into this buffer, starting at the
+     * given {@code start} index and at the given {@code index} in this buffer.
+     * The position of this buffer is unchanged.
+     *
+     * <p> In other words, an invocation of this method of the form
+     * <code>dst.put(index,&nbsp;csq,&nbsp;start,&nbsp;end)</code>
+     * has exactly the same effect as the loop
+     *
+     * <pre>{@code
+     *     for (int i = start, j = index; i < end; i++, j++)
+     *         dst.put(j, csq.charAt(i));
+     * }</pre>
```

Comment:
```suggestion
     * {@snippet lang=java :
     *     for (int i = start, j = index; i < end; i++, j++)
     *         dst.put(j, csq.charAt(i));
     * }
```

Reply:
Good suggestion. There are however eight other occurrences of the `<pre>{@code ...}</pre>` construct already in the file and I think I would prefer to change them all under a separate issue.